app/models/order.rb

```diff
@@ -4,11 +4,18 @@ class Order < ApplicationRecord
   has_many :order_items
 
   validates_presence_of :purchaser_name, :status
+  before_validation :save_purchaser_name
```

Comment:
Wondering about this as a use of callbacks--is it better to minimize callbacks, or to push logic into the model vs. the controller?